src/main/java/seedu/duke/helper/Validator.java

```diff
@@ -0,0 +1,197 @@
+        Pattern addressPattern = Pattern.compile("[\\w\\-\\s'()#]*");
+        Matcher addressMatcher = addressPattern.matcher(address);
+        return addressMatcher.matches();
+    }
+
+    private static boolean validateDob(String dobString) {
+        LocalDate dob;
+        try {
+            dob = LocalDate.parse(dobString);
+        } catch (DateTimeParseException dateTimeParseException) {
+            return false;
+        }
+        LocalDate today = LocalDate.now();
+        LocalDate dobLimit = LocalDate.parse("1900-01-01");
+        if (dob.isAfter(dobLimit) && dob.isBefore(today)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private static boolean validateAdmissionDate(String admissionDateString) {
+        LocalDate admissionDate;
+        try {
+            admissionDate = LocalDate.parse(admissionDateString);
+        } catch (DateTimeParseException dateTimeParseException) {
+            return false;
+        }
+        LocalDate today = LocalDate.now();
+        LocalDate admissionDateLimit = LocalDate.parse("1980-01-01");
+        if (admissionDate.isAfter(admissionDateLimit) && admissionDate.isBefore(today)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /* Validating person */
+    private static boolean validateAddPerson(String[] parameters) {
+        boolean isValid = true;
+        if (!validateNric(parameters[0])) {
+            UI.printParagraph("NRIC must start with a capital letter, "
+                    + "followed by 7 digits and end with a capital letter.");
+            isValid = false;
+        }
+        if (validateFullName(parameters[1])) {
+            UI.printParagraph("Full name must contain only alphabets and no special characters.");
+            isValid = false;
+        }
+        if (!validateAge(parameters[2])) {
+            UI.printParagraph("Age must be between 1 and 120 inclusive.");
+            isValid = false;
+        }
+        if (!validateGender(parameters[3])) {
+            UI.printParagraph("Gender must be a single character: M or F.");
+            isValid = false;
+        }
+        if (!validateAddress(parameters[4])) {
+            UI.printParagraph("Address must be alphanumeric. "
+                    + "Only these specific special characters are allowed: ' ( ) #");
+            isValid = false;
+        }
+        if (!validateDob(parameters[5])) {
+            UI.printParagraph("Date of birth must be in YYYY-MM-DD format. "
+                    + "It cannot be before 1900-01-01 or be today and after.");
+            isValid = false;
+        }
+        return isValid;
+    }
+
+    static boolean validateAddDoctor(String[] parameters) {
+        boolean isValid = validateAddPerson(Arrays.copyOfRange(parameters, 0, 6));
+
+        //validate full name cause specialization is also just a name
+        if (validateFullName(parameters[6])) {
```

Comment:
This one also, the current validateFullName returns true if the name is compliant. So it will currently make isValid false whenever a valid name is given.